src/emulator/mem/src/mem.cpp

```diff
@@ -81,14 +82,18 @@ bool init(MemState &state) {
     state.memory = Memory(static_cast<uint8_t *>(mmap(addr, length, prot, flags, fd, offset)), delete_memory);
 #endif
     if (!state.memory) {
+        LOG_CRITICAL("VirtualAlloc failed");
         return false;
     }
 
     state.allocated_pages.resize(length / state.page_size);
     const Address null_address = alloc(state, 1, "NULL");
     assert(null_address == 0);
 #ifdef WIN32
-    const BOOL res = VirtualProtect(state.memory.get(), state.page_size, PAGE_NOACCESS, nullptr);
+    DWORD lpflOldProtect;
```

Comment:
No Hungarian notation required. This prefix is wrong by the way, as lp = long pointer, which is what the & operator does. fl is flags I guess.

old_protect would match the existing code style.